json_serializable/test/src/extends_jsonkey_override.dart

```diff
@@ -0,0 +1,52 @@
+// @dart=3.8
+
+part of '_json_serializable_test_input.dart';
+
+// https://github.com/google/json_serializable.dart/issues/1437
+@ShouldGenerate(r'''
+CtorParamJsonKey _$CtorParamJsonKeyFromJson(Map<String, dynamic> json) =>
+    CtorParamJsonKey(
+      attributeOne: json['first'] as String,
+      attributeTwo: json['second'] as String,
+    );
+
+Map<String, dynamic> _$CtorParamJsonKeyToJson(CtorParamJsonKey instance) =>
+    <String, dynamic>{
+      'first': instance.attributeOne,
+      'second': instance.attributeTwo,
+    };
+''')
+@JsonSerializable()
+class CtorParamJsonKey {
+  CtorParamJsonKey({
+    @JsonKey(name: 'first') required this.attributeOne,
+    @JsonKey(name: 'second') required this.attributeTwo,
+  });
+
+  @JsonKey(name: 'fake_first')
```

Comment:
Ugh. Could we log when there are conflicts and tell users which one we're picking? Let me know if that's too much work...